warn/warn_bazel_api.go

```diff
@@ -832,3 +832,105 @@ func ruleImplReturnWarning(f *build.File) []*LinterFinding {
 
 	return findings
 }
+
+var legacyNamedParameters = map[string]string{
+	"all":     "elements",
+	"any":     "elements",
+	"tuple":   "x",
+	"list":    "x",
+	"len":     "x",
+	"str":     "x",
+	"repr":    "x",
+	"bool":    "x",
+	"int":     "x",
+	"dir":     "x",
+	"type":    "x",
+	"hasattr": "x",
+	"getattr": "x",
+	"select":  "x",
+}
```

Comment:
How are you going to fix functions with more than 1 argument?

Reply:
I was going to implement it in portions, but perhaps I should support flexible amount of arguments now and just add functions/methods later.

Reply:
Refactored to handle both legacy keyword and legacy positional parameters, even for the same function